### ISSUE TITLE
Compilation impossible because Canon EOS Digital Rebel XsI.icc is missing

### ISSUE THREAD

**Issue by Desmis** [Beep6581/RawTherapee#4508]
I want to merge dev with testoutputprofile, all is good except to generate Rawtherapee.exe
The compiler says me "cannot find :Canon EOS Digital Rebel XsI.icc

If I look in Github, branch dev, in rtdata/iccprofiles/input, no file of this type

Is I am missing someting ?

**Comment by Floessie** [comment]
The files in `rtdata` are gathered with a cmake glob pattern. Run `cmake [...]` in your build dir again, that should fix it.

HTH,
Flössie

**Comment by Desmis** [comment]
@Floessie 
Ok now it works, but for me it is Chineese :)
Excuse me, but as I said, I am a very poor "Computer scientist", and I don't understand why ? For All things with Cmake, etc.  I am as a monkey ! 

:)
jacques

**Comment by Desmis** [comment]
But should we update Rawpedia ?

**Comment by Beep6581** [comment]
@Desmis running `cmake` should pick up changes in the tree and update the list of things to compile, link, etc. In this case the tree changed and the old "recipe" was no longer valid. Re-running cmake is usually enough, but in extreme cases (e.g. switching from our GTK2 branch to the GTK3 branch) you might need to delete/rename the `build` folder.

I checked RawPedia and it seems fine to me. Can we close?

**Comment by Desmis** [comment]
Ok, you can close, but excuse my incompetence
* I do not see anywhere to say in which cases it is necessary to re-execute cmake
* why does we not add this files to rtdata by  ;  git add rtdata/xxxx ?

jacques

**Comment by Floessie** [comment]
@Desmis @Beep6581 The culprit is the use of `GLOB` in [`rtdata/CMakeLists.txt`](https://github.com/Beep6581/RawTherapee/blob/dev/rtdata/CMakeLists.txt#L2). When `cmake` is run it will pick up those files. But if a file is added or removed, CMake cannot detect that. This is a known ["problem"](https://stackoverflow.com/questions/32411963/why-is-cmake-file-glob-evil/32412044) with globbing, and that's why we list the files manually in [other](https://github.com/Beep6581/RawTherapee/blob/dev/rtengine/CMakeLists.txt#L30) `CMakeLists.txt` files. If we also do so with the resources in `rtdata/` the problem is solved.

This is not a `git` problem. :wink: 

I can work on this next week.

HTH,
Flössie

**Comment by Beep6581** [comment]
I deleted the file intentionally, it was a renamed copy of `Canon EOS 450D.icc`.

RawPedia instructions are for end users, and I want them to be simple to follow, fool-proof with no unnecessary branching, so they recommend always re-running CMake to avoid problems.

I updated the "Updating RawTherapee" section:
http://rawpedia.rawtherapee.com/Linux#Update_RawTherapee

For you as a developer, you should know that the build system should automatically detect changes to _source code_ and re-run CMake automatically, but _data_ changes do require re-running CMake manually because our data files are gathered with a CMake glob pattern, as @Floessie explained above, and the build system cannot know what changed.
https://cmake.org/cmake/help/latest/command/file.html

In conclusion, I re-run CMake always, because it only takes a second, and does not require me to know what changed.

**Comment by Desmis** [comment]
In conclusion, it is not obvious, your remarks prove it

But, now , I will always re-run Cmake :)

jacques

**Comment by Desmis** [comment]
Another question
I want to be considered a developer, but as I said before I'm in this kind of exercise (compilation, Link, ..) close to the monkey
When you say "RawPedia instructions are for end users", but if compilation failed, how they do ? they try as with a surprise pouch ?

**Comment by Beep6581** [comment]
@Desmis if users follow RawPedia instructions and [compilation fails](https://github.com/Beep6581/RawTherapee/issues?utf8=%E2%9C%93&q=is%3Aissue++label%3ACompilation+), they open an issue in GitHub or post in the forum.

Someone knowledgeable about the whole compilation process does not have to follow RawPedia verbatim, they can cut corners here and there if they know what they're doing. RawPedia assumes the reader does not know what they're doing, and so gives them one step-by-step path, and in fact recommends the automatic `build-rawtherapee` script.
The fewer choices, the lower the chance of a mistake.

**Comment by Desmis** [comment]
ok :)
